go-controller/pkg/cni/ovs.go

```diff
@@ -66,3 +70,46 @@ func ovsClear(table, record string, columns ...string) error {
 	_, err := ovsExec(args...)
 	return err
 }
+
+func ofctlExec(args ...string) (string, error) {
+	args = append([]string{"--timeout=30", "--no-stats", "--strict"}, args...)
+	var stdout, stderr bytes.Buffer
+	cmd := runner.Command("ovs-ofctl", args...)
+	cmd.SetStdout(&stdout)
+	cmd.SetStderr(&stderr)
+
+	cmdStr := strings.Join(args, " ")
+	klog.V(5).Infof("exec: ovs-ofctl %s", cmdStr)
+
+	err := cmd.Run()
+	if err != nil {
+		stderrStr := stderr.String()
+		klog.Errorf("exec: ovs-ofctl %s : stderr: %q", cmdStr, stderrStr)
+		return "", fmt.Errorf("failed to run 'ovs-ofctl %s': %v\n  %q", cmdStr, err, stderrStr)
+	}
+	stdoutStr := stdout.String()
+	klog.V(5).Infof("exec: ovs-ofctl %s: stdout: %q", cmdStr, stdoutStr)
+
+	trimmed := strings.TrimSpace(stdoutStr)
+	// If output is a single line, strip the trailing newline
+	if strings.Count(trimmed, "\n") == 0 {
+		stdoutStr = trimmed
+	}
+	return stdoutStr, nil
+}
+
+func waitForPodFlows(mac string) error {
+	return wait.PollImmediate(200*time.Millisecond, 20*time.Second, func() (bool, error) {
+		// Query the flows by mac address
+		query := fmt.Sprintf("table=9,dl_src=%s", mac)
```

Comment:
well we want port security, because we do not want the flow to show up until the end of addLogicalPort. Is table 8 port security as well? I'll bring up a cluster and see what it is.

Reply:
@girishmg @trozet from week before last:

```
(10:20:56 AM) mmichelson: dcbw, assuming you always set port security on your logical switch ports, then table 9 is a reliable one to search for the pod IP in.
(10:21:07 AM) mmichelson: Because table 9 is where port security is performed.
```